src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let str = '';
+    let count = 0;
+
+    for (let ch of this) {
+      count++;
+
+      if (ch === undefined || ch === null) {
```

Comment:
Instead of checking if ch is undefined or null, you can check if this[i] is undefined or null.